src/object/cli_ec.c

```diff
@@ -57,27 +57,67 @@ struct ec_params {
 	struct ec_params        *next;	/* Pointer to next entry in list. */
 };
 
+struct ec_fetch_params {
+	daos_iod_t		*iods;	/* Replaces iod array in fetch. */
+	unsigned int		 nr;	/* number of records in iods    */
+	daos_iod_t			 niod;
+	struct ec_fetch_params		*next;/* Next entry in list */
+};
+
+static bool
+ec_is_full_stripe(daos_iod_t *iod, struct daos_oclass_attr *oca,
+		  unsigned int recx_idx)
+{
+	unsigned int	ss = oca->u.ec.e_k * oca->u.ec.e_len;
+	unsigned int start = iod->iod_recxs[recx_idx].rx_idx * iod->iod_size;
+	unsigned int length = iod->iod_recxs[recx_idx].rx_nr * iod->iod_size;
+
+	if (length < ss && start/ss == (start+length)/ss) {
+		return false;
+	}
+	if (start % ss) {
+		unsigned long so = ss - start % ss;
+
+		start += so;
+		length -= so;
+	}
+
+	if (length < ss)
+		return false;
+	return true;
+}
+
 /* Determines weather a given IOD contains a recx that is at least a full
  * stripe's worth of data.
  */
 static bool
-ec_has_full_stripe(daos_iod_t *iod, struct daos_oclass_attr *oca,
-		   uint64_t *tgt_set)
+ec_has_full_or_mult_stripe(daos_iod_t *iod, struct daos_oclass_attr *oca,
+			   uint64_t *tgt_set)
 {
 	unsigned int	ss = oca->u.ec.e_k * oca->u.ec.e_len;
 	unsigned int	i;
 
 	for (i = 0; i < iod->iod_nr; i++) {
 		if (iod->iod_type == DAOS_IOD_ARRAY) {
-			int start = iod->iod_recxs[i].rx_idx * iod->iod_size;
-			int length = iod->iod_recxs[i].rx_nr * iod->iod_size;
+			unsigned int start =
+				iod->iod_recxs[i].rx_idx * iod->iod_size;
+			unsigned int length =
+				iod->iod_recxs[i].rx_nr * iod->iod_size;
 
-			if (length < ss) {
+			if (length < ss && start/ss == (start+length)/ss) {
```

Comment:
start/ss == (start+length)/ss 
I might not be thinking straight, but wouldn't that mean length == 0?

Reply:
It would mean that start + length is in the same stripe.  Integer division gives the floor

Reply:
What  == testing is whether a recx both starts and ends within the same stripe. If the the integer divisions are equal, than both the start of the recx and the end of the recx lie within the same stripe.  Since length is less than stripe size, it's not a full stripe. Since the start and end are in the same stripe, it's not a multi-stripe. Thus we conclude that this recx is not a full or multi stripe and continue (move on the the next recx).

Reply:
There are other ways to determine whether a start/length falls in the same stripe, but this one is intuitive to me.

Reply:
duh ... yes, that makes sense. I wasn't thinking straight :-P